### PR TITLE
V5 Removing native js dependencies

### PR DESCRIPTION
**Pull request by nebojsa94**
This PR removes keccak and secp256k1 dependencies and instead uses ethereum-cryptography package that doesn't require native dependency compiling.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,13 @@
-const createKeccakHash = require('keccak')
-const secp256k1 = require('secp256k1')
+const { keccak224, keccak384, keccak256: k256, keccak512 } = require('ethereum-cryptography/keccak')
+const {
+  privateKeyVerify,
+  publicKeyCreate,
+  publicKeyVerify,
+  publicKeyConvert,
+  sign,
+} = require('ethereum-cryptography/shims/hdkey-secp256k1v3')
```
Comment (alcuadrado) [inline index.js:8]: See: https://github.com/ethereumjs/ethereumjs-util/pull/259/files#r441088580

File: index.js [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
+const { ecdsaRecover } = require('ethereum-cryptography/secp256k1')
+const secp256k1 = require('ethereum-cryptography/secp256k1')
```
Comment (alcuadrado) [inline index.js:10]: https://github.com/ethereumjs/ethereumjs-util/pull/259/files#r441090419

File: index.js [hunk 3]
```diff
@@ -230,7 +238,23 @@ exports.keccak = function (a, bits) {
...
+      return k256(a)
+    }
+    case 384: {
+      return keccak384(a)
+    }
+    case 512: {
+      return keccak512(a)
+    }
+    default: {
+      throw new Error(`Invald algorithm: keccak${bits}`)
```
Comment (alcuadrado) [inline index.js:255]: See: https://github.com/ethereumjs/ethereumjs-util/pull/259/files#r441089259

**Comment by alcuadrado** [comment]
This comment is also relevant here: https://github.com/ethereumjs/ethereumjs-util/pull/261#issuecomment-656753003

**Comment by nebojsa94** [comment]
Here are a couple of PRs that are testing this code:
https://github.com/Tenderly/eth-sig-util/pull/1
https://github.com/Tenderly/merkle-patricia-tree/pull/1

**Pull request merged**